cmd/cli/kubectl-kyverno/commands/test/test.go

```diff
@@ -98,6 +98,52 @@ func runTest(out io.Writer, testCase test.TestCase, registryAccess bool, auditWa
 	} else {
 		fmt.Fprintln(out, "  Applying", len(policies)+len(vaps), pluralize.Pluralize(len(policies)+len(vaps), "policy", "policies"), "to", len(uniques), pluralize.Pluralize(len(uniques), "resource", "resources"), "...")
 	}
+	policyToUnknownRules := make(map[string]string)
+	unknwonPolicies := make(map[string]string)
+	for _, res := range testCase.Test.Results {
+		if res.IsValidatingAdmissionPolicy {
+			continue
+		}
+		var ns string
+		var n bool
+		r := false
+		for _, policy := range policies {
+			if policy.IsNamespaced() {
+				ns = policy.GetNamespace()
+			}
+			if res.Policy == policy.GetName() || res.Policy == "default/"+policy.GetName() || res.Policy == ns+"/"+policy.GetName() {
+				n = true
+				for _, rule := range autogen.ComputeRules(policy) {
+					if res.Rule == rule.Name {
+						r = true
+						break
+					}
+				}
+				break
+			}
+		}
```

Comment:
It can be re-written as follows:
```
for _, policy := range policies {
  if policy.IsNamespaced() {
      ns = policy.GetNamespace()
  }

  // ensure that the policy specified in the results matches one of the policies provided in the given files.
  if res.Policy != policy.GetName() && res.Policy != "default/"+policy.GetName() && res.Policy != ns+"/"+policy.GetName() {
      continue
  }

  policyExists = true

  // ensure that the rule specified in the results exists in the policy.
  for _, rule := range autogen.ComputeRules(policy) {
       if res.Rule == rule.Name {
          ruleExists = true
          break
       }
  }
  break
}
```